appcenter-data/android/src/main/java/com/microsoft/appcenter/reactnative/data/AppCenterReactNativeDataUtils.java

```diff
@@ -64,4 +69,68 @@ public static WritableArray convertJsonArrayToWritableArray(JsonArray jsonArray)
         }
         return writableArray;
     }
+
+    public static JsonObject convertReadableMapToJsonObject(ReadableMap readableMap) {
+        JsonObject jsonObject = new JsonObject();
+        if (readableMap == null) {
+            return jsonObject;
+        }
+        ReadableMapKeySetIterator iterator = readableMap.keySetIterator();
+        while (iterator.hasNextKey()) {
+            String key = iterator.nextKey();
+            ReadableType type = readableMap.getType(key);
+            switch (type) {
+                case Number:
+                    jsonObject.addProperty(key, readableMap.getDouble(key));
+                    break;
+                case String:
+                    jsonObject.addProperty(key, readableMap.getString(key));
+                    break;
+                case Boolean:
+                    jsonObject.addProperty(key, readableMap.getBoolean(key));
+                    break;
+                case Map:
+                    jsonObject.add(key, convertReadableMapToJsonObject(readableMap.getMap(key)));
+                    break;
+                case Array:
+                    jsonObject.add(key, convertReadableArrayToJsonArray(readableMap.getArray(key)));
+                    break;
+                default:
+                    jsonObject.add(key, JsonNull.INSTANCE);
+                    break;
+            }
+        }
+        return jsonObject;
+    }
+
+    public static JsonArray convertReadableArrayToJsonArray(ReadableArray readableArray) {
+        JsonArray jsonArray = new JsonArray();
+        if (readableArray == null) {
+            return jsonArray;
+        }
+        for (int i = 0; i < readableArray.size(); i++) {
+            ReadableType type = readableArray.getType(i);
+            switch (type) {
+                case Number:
+                    jsonArray.add(readableArray.getDouble(i));
+                    break;
+                case String:
+                    jsonArray.add(readableArray.getString(i));
+                    break;
+                case Boolean:
+                    jsonArray.add(readableArray.getBoolean(i));
+                    break;
+                case Map:
+                    jsonArray.add(convertReadableMapToJsonObject(readableArray.getMap(i)));
+                    break;
+                case Array:
+                    jsonArray.add(convertReadableArrayToJsonArray(readableArray.getArray(i)));
+                    break;
+                default:
+                    jsonArray.add(JsonNull.INSTANCE);
```

Comment:
Currently, the native Android SDK does not support serialization of null fields. A separate work item will be created to update the native SDK to enable this feature.